packages/flame/lib/src/components/mixins/draggable.dart

```diff
@@ -84,7 +84,7 @@ mixin HasDraggableComponents on BaseGame {
   }
 
   void _onGenericEventReceived(bool Function(Draggable) handler) {
-    for (final c in components.toList().reversed) {
+    for (final c in components.reversed()) {
```

Comment:
I think I might have written this initially, but we should probably just iterate over the list in reverse instead of creating a new reversed list to iterate over.

Reply:
that does exactly the same thing as before

Reply:
That's what I'm saying, that it was inefficient before too and that we should consider not creating a new list every time an event is fired. So either maintain another list in the reverse order inside of QOS, or iterate it from the end.

Reply:
ah gotcha, let me see how to iterate properly. I think that was the reason I "extracted" to a method but then I forgot to look into it

Reply:
I don't think it's possible, sadly. unless we keep a secondary copy of the entire set in sync with negative priorities. you can't reverse an interator just by the nature of the interface that it exposes. you have to first materialize the whole list

Reply:
Okay, let's handle this later if it becomes a bottle neck.